packages/lesswrong/server/manualMigrations/2023-01-10-revertSubforumNotifSettings.ts

```diff
@@ -0,0 +1,104 @@
+       */
+
+      // Based on when the PRs were deployed. There is the possibility for some edge cases here, but as a matter of fact there aren't any
+      // rows that close to either of these times so it's fine.
+      const notifsEnabledDatetime = '2022-12-05 13:47:00'
+      const notifsDisabledDatetime = '2022-12-24 17:41:00'
+      
+      /**
+       * This query filters by the following criteria:
+       * - The subscription is for a subforum, with state = 'subscribed' (rather than 'supressed')
+       * - It was created between the dates when the notifications were enabled by default
+       * - There is only one subscription for the user and document, meaning (probably) that the user was auto-subscribed by joining and hasn't subsequently unsubscribed
+       *   - Edge case: The user was already deliberately unsubscribed from the tag before the subforum existed, and then joined the subforum which caused them to subscribe.
+       *     I have checked the cases where there is more than one subscription row and in the only case like this they were already *subscribed* before the subforum was created,
+       *     so it's fine to leave that one
+       */
+      const matchingSubscriptionsSql = `
+      SELECT
+        subscription_ids
+      FROM (
+        SELECT
+          count(*) AS total_count,
+          max("createdAt") AS most_recent_change,
+          array_agg("state") as subscribed_state,
+          array_agg("deleted") as deleted_state,
+          array_agg("_id") as subscription_ids,
+          "userId",
+          "documentId"
+        FROM
+          "Subscriptions"
+        WHERE
+          "documentId" IN (${subforumTagIds.map(id => `'${id}'`).join(",")}) -- forecasting, software-engineering, effective-giving, bioethics
+        GROUP BY
+          "userId",
+          "documentId") A
+        WHERE
+          total_count = 1 AND most_recent_change > '${notifsEnabledDatetime}' AND most_recent_change < '${notifsDisabledDatetime}' AND
+          '{subscribed}' @> subscribed_state AND '{f}' @> deleted_state
+      ORDER BY
+        most_recent_change DESC, "documentId";
+      `
+      const matchingSubscriptions = await db.any(matchingSubscriptionsSql)
+      const matchingSubscriptionIds = matchingSubscriptions.map(sub => sub.subscription_ids[0])
+      
+      console.log(`Found ${matchingSubscriptionIds.length} subscriptions to soft delete`)
+      if (matchingSubscriptionIds.length < 25) { // I happen to know there are only 11, so it's useful to have this in the logs
+        console.log("Subscription ids:", matchingSubscriptionIds)
+      }
+
+      // set deleted to true for all of them (only soft delete in case we want to do future data munging related to this)
+      await Subscriptions.rawUpdateMany({_id: {$in: matchingSubscriptionIds}}, {$set: {deleted: true}}, {multi: true})
```

Comment:
I don't think we need `{multi: true}` here?

Reply:
Yeah I guess `UpdateMany` takes care of this